src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -161,7 +161,9 @@ private function guessHandledClasses(\ReflectionClass $handlerClass, string $ser
     private function registerReceivers(ContainerBuilder $container)
     {
         $receiverMapping = array();
-        foreach ($container->findTaggedServiceIds($this->receiverTag) as $id => $tags) {
+        $taggedReceivers = $container->findTaggedServiceIds($this->receiverTag);
+
+        foreach ($taggedReceivers as $id => $tags) {
             foreach ($tags as $tag) {
                 $receiverMapping[$id] = new Reference($id);
```

Comment:
It's unrelated to this PR, but I believe that this statement can be moved to the parent loop.

Reply:
I've updated it 😉